### PR TITLE
Integration of new event listener

### PR DESCRIPTION
**Pull request by sandstromviktor**
## Description

This PR integrates new [Event Listener](https://github.com/ScilifelabDataCentre/serve-event-listener) into serve. 
Checkout the repo for details on how it works. 
I have a [linked PR](https://github.com/ScilifelabDataCentre/serve-charts/pull/19) in the serve-charts repo. Please review this as well. 

I've also fixed some subdomain bugs, where changing subdomains caused issues. 

solves 
- [SS-764](https://scilifelab.atlassian.net/browse/SS-764)
- [SS-854](https://scilifelab.atlassian.net/browse/SS-854) 
- [SS-838](https://scilifelab.atlassian.net/browse/SS-838)
- [SS-868](https://scilifelab.atlassian.net/browse/SS-868)

## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [X] I have included a link to the issue on GitHub or JIRA (if any)
- [ ] I have included migration files (if there are changes to the model classes)
- [X] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [X] I have updated the related documentation (if necessary)
- [ ] I have updated the release notes (releasenotes.md)
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [X] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

### CODE REVIEW

File: apps/views.py [hunk 1]
```diff
@@ -274,23 +274,19 @@ def update_app_instance(self, request, project, appinstance, app_settings, body)
     def update_resource(self, request, appinstance, current_release_name):
         domain = appinstance.parameters["global"]["domain"]
         # if subdomain is set as --generated--, then use appname
-        if request.POST.get("app_release_name") == "":
-            new_release_name = appinstance.parameters["appname"]
-        else:
+        if request.POST.get("app_release_name"):
             new_release_name = request.POST.get("app_release_name")
+        else:
+            new_release_name = appinstance.parameters["appname"]
```
Comment (churnikov) [inline apps/views.py:280]:
```suggestion
        new_release_name =  request.POST.get("app_release_name", appinstance.parameters["appname"])
```

**Review by churnikov — COMMENTED** [review]
Could you please describe in this task how to reproduce this bug?
https://scilifelab.atlassian.net/browse/SS-854

**Comment by churnikov** [comment]
Tried to run it locally, but delete doesn't seem to work for me. Apps just get stuck in "Delete..." state

**Comment by churnikov** [comment]
@akochari Could you please write here or add confluence link regarding issues with subdomains names?

File: docker-compose.yaml [hunk 2]
```diff
@@ -48,6 +48,11 @@ services:
       internal_network:
         aliases:
           - studio.127.0.0.1.nip.io
+    healthcheck:
+      test: ["CMD", "wget", "--quiet", "--tries=1", "--spider", "http://localhost:8080"]
```
Comment (churnikov) [inline docker-compose.yaml:52]: You can do that now, I guess
```suggestion
      test: ["CMD", "wget", "--quiet", "--tries=1", "--spider", "http://localhost:8080/api/are-you-there/"]
```

File: docker-compose.yaml [hunk 3]
```diff
@@ -29,6 +29,7 @@ services:
       - STUDIO_STORAGECLASS=microk8s-hostpath
       - STUDIO_ACCESSMODE=ReadWriteMany
       - DJANGO_SUPERUSER_PASSWORD=dGhpaXNhdmVyeW5vdHNhZmVvbmx
+      - EVENT_USER_PASSWORD=dGhpaXNhdmVyeW5vdHNhZmVvbmx
```
Comment (churnikov) [inline docker-compose.yaml:32]: We should add this to argo as well as I understand it

File: docker-compose.yaml [hunk 4]
```diff
@@ -151,7 +152,7 @@ services:
     environment:
       - KUBECONFIG=/cluster.conf
       - DEBUG=True
-      - USERNAME=admin@test.com
+      - USERNAME=event_user
```
Comment (churnikov) [inline docker-compose.yaml:155]:
```suggestion
      - USERNAME=event_user@test.com
```
Shouldn't it be like this?

File: docker-compose.yaml (hunk 3 above)
Comment (sandstromviktor) [reply docker-compose.yaml:32]: Yes! In serve charts, this is set as a random secret. This is just for development.

File: docker-compose.yaml (hunk 4 above)
Comment (sandstromviktor) [reply docker-compose.yaml:155]: Yes, thanks!

**Comment by alfredeen** [comment]
I have tested creating, renaming, and deleting a jupyter lab instance. The status update works and finishes with status "Deleted".

**Review by alfredeen — APPROVED** [review]
Nice work! I have tested creating and deleting apps and a jupyter notebook and the status updates work.

**Review by churnikov — APPROVED** [review]

**Pull request merged**